qlty-check/src/cache.rs

```diff
@@ -91,6 +93,8 @@ pub struct IssuesCacheHit {
 
 #[derive(Debug, Clone)]
 pub struct IssuesCacheKey {
+    repository_sha: Option<String>,
```

Comment:
We may want to clarify this is the root tree SHA (not a commit SHA, which is more common)

Reply:
I'm not really sure what the difference is. The commit sha should match the tree in the case of clean file nodes in the workspace.

Reply:
In terms of naming, the commit SHA and the tree SHA are different values, so that is the potential confusion that I think it would be valuable to resolve.

From reading the code in this PR, it looks like we are using the tree SHA rather than the commit SHA but the libgit2 calls don't make it entirely clear.

Functionally, the tree SHA will behave marginally better than the commit SHA because there are a number of cases like merges which roll the commit SHA but don't change the tree.